bootstrap/bootstrap_test.go

```diff
@@ -78,35 +79,44 @@ func TestGetValuesToRedactEmpty(t *testing.T) {
 	assert.Equal(t, 0, len(valuesToRedact))
 }
 
-func TestStartTracing(t *testing.T) {
-	oriCtx := context.Background()
+func TestStartTracing_NoTracingBackend(t *testing.T) {
 	var err error
 
-	// When there's no Datadog tracing address, the tracer should be a no-op.
-	cfg := Config{}
+	// When there's no tracing backend, the tracer should be a no-op.
 	b := New(Config{})
+
+	oriCtx := context.Background()
 	b.shell, err = shell.NewWithContext(oriCtx)
-	if err != nil {
-		assert.FailNow(t, "Unexpected error while createing shell: %v", err)
-	}
-	span, ctx, stopper := b.startTracing(oriCtx)
+	assert.NoError(t, err)
+
+	span, _, stopper := b.startTracing(oriCtx)
+	assert.Equal(t, span, &tracetools.NoopSpan{})
+	span.FinishWithError(nil) // Finish the nil span, just for completeness' sake
+
+	// If you call opentracing.DefaultTracer() without having set it first, it returns a NoopTracer
+	// In this test case, we haven't touched opentracingt at all, so we get the NoopTracer
```

Comment:
The comment doesn't match the code here, should this be GlobalTracer?
```suggestion
	// If you call opentracing.GlobalTracer() without having set it first, it returns a NoopTracer
	// In this test case, we haven't touched opentracing at all, so we get the NoopTracer
```

Reply:
yep, you're right - this code was extracted from when there was also OpenTelemetry stuff in here. In Otel, it's called the DefaultTracer. Will shuffle this around to use the right nomenclature